src/mysql_protocol.erl

```diff
@@ -210,9 +210,19 @@ fetch_query_response(SockModule, Socket, AllowedPaths, FilterMap, Timeout) ->
 %% @doc Prepares a statement.
 -spec prepare(iodata(), module(), term()) -> #error{} | #prepared{}.
 prepare(Query, SockModule, Socket) ->
+    prepare(Query, SockModule, Socket, infinity).
```

Comment:
Have you considered changing this to a hard-coded value such as `?cmd_timeout`, which is used for BEGIN, COMMIT and ROLLBACK? Do you think it would break backward compatibility to do that?

Reply:
I am not sure if applications have hard dependencies there. Thinking through this again, maybe we can get away with breaking things by announcing it under some page easily visible (for changelog) and then leaving it for others to find during their testing if you did blind upgrades.

Reply:
Yes, it's an interesting question. If we set a quite high number such as 10 seconds, I don't think it will break any applications. It will timeout only if the database is overloaded, like in your case, and then it's good that it times out.

How many seconds are you planning to use for your application? Not yet decided?